src/libraries/System.Runtime/tests/System.IO.Tests/Stream/Stream.NullTests.cs

```diff
@@ -196,14 +198,93 @@ public static async Task TestCanceledNullTextReaderAsync(TextReader input)
+            output.Write(" {0} {1}  {2}    {3}", (Int128)4, (UInt128)5, (nint)6, (nuint)7);
+            output.WriteLine();
+            output.WriteLine(true);
+            output.WriteLine('a');
+            output.WriteLine((char[])null);
+            output.WriteLine(new char[] { 'b', 'c' });
+            output.WriteLine(42m);
+            output.WriteLine(43d);
+            output.WriteLine(44f);
+            output.WriteLine(45);
+            output.WriteLine(46L);
+            output.WriteLine(DayOfWeek.Monday);
+            output.WriteLine((string)null);
+            output.WriteLine("Tuesday");
+            output.WriteLine((StringBuilder)null);
+            output.WriteLine(new StringBuilder("Wednesday"));
+            output.WriteLine(47u);
+            output.WriteLine(48ul);
+            output.WriteLine("Thursday".AsSpan());
+            output.WriteLine(" {0} ", "Friday");
+            output.WriteLine(" {0}{1} ", "Saturday", "Sunday");
+            output.WriteLine(" {0} {1}  {2}", TimeSpan.FromSeconds(1), TimeSpan.FromMinutes(2), TimeSpan.FromDays(3));
+            output.WriteLine(" {0} {1}  {2}    {3}", (Int128)4, (UInt128)5, (nint)6, (nuint)7);
+            Assert.True(output.WriteAsync('a').IsCompletedSuccessfully);
+            Assert.True(output.WriteAsync((char[])null).IsCompletedSuccessfully);
+            Assert.True(output.WriteAsync(new char[] { 'b', 'c' }).IsCompletedSuccessfully);
+            Assert.True(output.WriteAsync((string)null).IsCompletedSuccessfully);
+            Assert.True(output.WriteAsync("Tuesday").IsCompletedSuccessfully);
+            Assert.True(output.WriteAsync((StringBuilder)null).IsCompletedSuccessfully);
+            Assert.True(output.WriteAsync(new StringBuilder("Wednesday")).IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync().IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync('a').IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync((char[])null).IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync(new char[] { 'b', 'c' }).IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync((string)null).IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync("Tuesday").IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync((StringBuilder)null).IsCompletedSuccessfully);
+            Assert.True(output.WriteLineAsync(new StringBuilder("Wednesday")).IsCompletedSuccessfully);
+
+            if (output is StreamWriter sw)
+            {
+                Assert.False(sw.AutoFlush);
+                sw.AutoFlush = true;
+                Assert.False(sw.AutoFlush);
+
+                Assert.Same(Stream.Null, sw.BaseStream);
+            }
+
+            // Use some parallelism in an attempt to validate statelessness
+            string longLine = new string('#', 100_000);
+            Parallel.For(0, 25, i => output.WriteLine(longLine));
```

Comment:
Is it going to work fine for WASM?

Reply:
I believe so, but it will just end up running it serially.